packages/checkout/shipping/shipping-rates-control/shipping-rates-control.js

```diff
@@ -87,22 +104,54 @@ const ShippingRatesControl = ( {
 			) }
 			showSpinner={ true }
 		>
-			<Packages
+			<ExperimentalOrderShippingPackages.Slot
 				className={ className }
-				collapsible={
-					shippingRates.length > 1 && collapsibleWhenMultiple
-				}
+				collapsibleWhenMultiple={ collapsibleWhenMultiple }
 				noResultsMessage={ noResultsMessage }
 				renderOption={ renderOption }
-				shippingRates={ shippingRates }
+			/>
+			<Packages
+				packages={ shippingRates }
+				noResultsMessage={ noResultsMessage }
 			/>
 		</LoadingMask>
 	);
 };
 
+/**
+ * Renders multiple packages within the slotfill.
+ *
+ * @param {Object} props Incoming props.
+ * @param {Array} props.packages Array of packages.
+ * @param {React.ReactElement} props.noResultsMessage Rendered when there are no packages.
+ * @return {React.ReactElement|Array} Rendered components.
+ */
+const Packages = ( { packages, noResultsMessage } ) => {
+	const { selectShippingRate, selectedShippingRates } = useSelectShippingRate(
+		packages
+	);
+
+	if ( ! packages.length ) {
+		return noResultsMessage;
+	}
+
+	return packages.map( ( { package_id: packageId, ...packageData } ) => (
+		<ExperimentalOrderShippingPackages key={ packageId }>
+			<Package
+				key={ packageId }
+				packageData={ packageData }
+				onSelectRate={ ( newShippingRate ) => {
+					selectShippingRate( newShippingRate, packageId );
+				} }
+				selected={ selectedShippingRates[ packageId ] }
+			/>
+		</ExperimentalOrderShippingPackages>
+	) );
```

Comment:
I'm not sure this code should live in our public package, or if it does make sense to be here. Seeing as this is calling a slotFill, I'd prefer if it's in our internal code.

Reply:
It wasn't prior to some of the other refactors. Does the entire control need to be in this public package? If we move it, where else is suitable?

Reply:
I think we moved it while trying to fix some circular dependency issue but I think we can return it if we're not exposing it to extensions if not, the least thing we  can do is move back the Slot part back to `base/components/cart-checkout`.